### PR TITLE
Update the p3 pre- and post- ambles to facilitate coupling with other processes

### PR DESCRIPTION
**Pull request by AaronDonahue**
This PR updates the pre- and post- ambles in P3 microphysics so that variables expected by other atmosphere processes are in the appropriate format. 

These changes also reflect decisions made about which variables will be carried in the field manager, and which variables will be computed on the fly in the pre-amble of a process.

Finally this PR will remove unused variables from P3, addressing issue #959 and #966.  Note, we have decided to keep these variables in the F90 code, as a result, this PR introduces a structure to the P3 interface that will hold "deprecated" variables that are no longer used in the cpp implementation but must be kept as long as our testing call F90 code, or until these variables are removed from the F90 code.

### CODE REVIEW

File: components/scream/src/physics/p3/atmosphere_microphysics.cpp
Comment (PeterCaldwell) [inline components/scream/src/physics/p3/atmosphere_microphysics.cpp:245]: If I understand correctly, you're still outputting these variables we don't want. Do we really need to do this? I thought Conrad's strategy was to still produce these variables in F90, but not to actually compare them against anything anymore... so the C++ wouldn't need to carry them anymore. Am I missing something?

File: components/scream/src/physics/p3/p3_functions_f90.cpp
Comment (AaronDonahue) [inline components/scream/src/physics/p3/p3_functions_f90.cpp:3270]: @PeterCaldwell , the following changes address this brittle-ness, I think.  I can delete this comment from you if you think that it isn't brittle anymore.

**Comment by AaronDonahue** [comment]
Since this PR removes variables from the p3-output, I think it is expected that `p3_run_and_cmp` will fail when compared with baselines from master.

**Comment by AaronDonahue** [comment]
@PeterCaldwell , when you have a chance could you take one more look at this PR to verify that it has addressed your questions?  I also had one question for you about a comment of yours in the code that I might be able to delete now, see above.

File: components/scream/src/physics/p3/p3_functions_f90.cpp
Comment (PeterCaldwell) [reply components/scream/src/physics/p3/p3_functions_f90.cpp:3270]: Looks good.

File: components/scream/src/physics/p3/p3_iso_c.f90
Comment (PeterCaldwell) [inline components/scream/src/physics/p3/p3_iso_c.f90:144]: Isn't nevapr supposed to be deleted everywhere?

Comment (AaronDonahue) [reply components/scream/src/physics/p3/p3_iso_c.f90:144]: oh yes, I think this PR was opened before the cleanup PR went in.  I can clean that up

Comment (AaronDonahue) [reply components/scream/src/physics/p3/p3_iso_c.f90:144]: it isn't deleted in the F90 code, right?  At least not in this PR.  I can go ahead and make that change too if you would like.

**Review by PeterCaldwell — APPROVED** [review]
Looks good, thanks Aaron!

**Comment by AaronDonahue** [comment]
I had to rebase this branch because of conflicts with master.  I'll let the pre-merge checks go through and then go ahead and manually merge once I verify that the p3_run_and_cmp tests are still the only ones that fail.

**Comment by AaronDonahue** [comment]
update: they didn't pop up as conflicts so I didn't see them until now, but there was a change to how we add fields in a processes that caused the build failures.  I'll fix this now.

**Review by PeterCaldwell — APPROVED** [review]

**Comment by AaronDonahue** [comment]
@bartgol , I'm not sure why the pre-merge inspection is still pending for this PR.  Peter definitely reapproved it as the last action on the PR so presumably this check should be done.  I can't merge the pull request until it is.  Am I missing something?

**Comment by bartgol** [comment]
> @bartgol , I'm not sure why the pre-merge inspection is still pending for this PR. Peter definitely reapproved it as the last action on the PR so presumably this check should be done. I can't merge the pull request until it is. Am I missing something?

The pre-merge inspection is performed _after_ testing passed. As opposed to the pre-test inspection (which is usually skipped since we're part of the scream team), which happens before testing.

Once tests pass, the AT will do the pre-merge inspection, which amounts to check whether there is a review approval for the current tip of the branch _and_ no reviews that requested changes.

**Comment by AaronDonahue** [comment]
ahhh, ok.  Sorry I misunderstood.  Tests won't pass.  I'm guessing I don't have the ability to override the AT and merge a PR.  Are you able to do that @bartgol?

Test fail explanation: This PR changes the arguments to P3 by removing a few, we expect p3_run_and_cmp to fail, which it does.

**Pull request merged**
